docs/reference/client-side-encryption.txt

```diff
@@ -523,6 +546,50 @@ keys are stored in the ``admin`` database in the ``datakeys`` collection:
 
 There is no default key vault namespace, and this option must be provided.
 
+``:kms_providers``
+~~~~~~~~~~~~~~~~~~
+A Hash that contains KMP provider names as keys, and provider options as values.
+
+.. code-block:: ruby
+
+  Mongo::Client.new(['localhost:27017],
+    auto_encryption_options: {
+      key_vault_namespace: 'admin.datakeys',
+      kms_providers: {
+        aws: {
+          access_key_id: 'IAM-ACCESS-KEY-ID',
+          secret_access_key: 'IAM-SECRET-ACCESS-KEY'
+        }
+      }
+    }
+  )
+
+``:kms_tls_options``
+~~~~~~~~~~~~~~~~~~
+A Hash that contains KMP provider names as keys, and TLS options to connect to
+corresponding providers.
+
+.. code-block:: ruby
+
+  Mongo::Client.new(['localhost:27017],
+    auto_encryption_options: {
+      key_vault_namespace: 'admin.datakeys',
+      kms_providers: {
+        kmip: {
+          endpoint: 'KMIP-SERVER-HOST'
+        }
+      },
+      kms_tls_options: {
+        kmip: {
+          ssl_ca_cert: 'PATH-TO-CA-FILE',
+          ssl_cert: 'PATH-TO-CLIENT-CERT-FILE',
+          ssl_key: 'PATH-TO-CLIENT-KEY-FILE'
```

Comment:
We have an open ticket to rename ssl prefixes to tls. Do you have an idea how feasible it would be to use the tls prefixes in the FLE changes you've been making recently?

Reply:
The spec says `The TLS options SHOULD be consistent with the existing TLS options for MongoDB server TLS connections.` I understand this as we should have the same set of TLS options for our client, and for KMIP.

Reply:
Let's punt on the renaming and do it all together later.